spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -277,7 +296,9 @@ public void startAckThread() {
 
 	protected void send(DatagramPacket packet) throws Exception {
 		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(this.getDestinationAddress());
```

Comment:
Right, only in the affected places.
This isn't critical like typo in log message, so we do this kind of code style fixes only during the original fix for an issue.